cmd/skaffold/app/cmd/debug.go

```diff
@@ -21,26 +21,54 @@ import (
 	"io"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 
 	debugging "github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy"
 )
 
+// for tests
+var doDebug = runDebug
+
 // NewCmdDebug describes the CLI command to run a pipeline in debug mode.
+// Unlike `dev`, `debug` defaults `auto-build` and `auto-deploy` to `false`.
 func NewCmdDebug() *cobra.Command {
+	// local copies to avoid aliasing references from `NewCmdDev` https://github.com/GoogleContainerTools/skaffold/issues/4129
+	var trigger string
+	var autoBuild, autoDeploy, autoSync bool
+	var targetImages []string
+	var watchPollInterval int
+
 	return NewCmd("debug").
 		WithDescription("[beta] Run a pipeline in debug mode").
 		WithLongDescription("Similar to `dev`, but configures the pipeline for debugging.").
 		WithCommonFlags().
-		NoArgs(doDebug)
+		WithFlags(func(f *pflag.FlagSet) {
+			f.StringVar(&trigger, "trigger", "notify", "How is change detection triggered? (polling, notify, or manual)")
```

Comment:
Can those flags be added to the common flags list, now that they are used by 2 commands?

Reply:
oh right 🤦